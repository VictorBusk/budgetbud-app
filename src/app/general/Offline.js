import React, {Component} from 'react';
import {View, Image} from 'react-native';
import {Container, Label} from "native-base";
import {container, text} from "../../style/index";
import I18n from "../../strings/i18n";

class Offline extends Component {
    render() {
        return (
            <Container style={container.signedOutContainer}>
                <Image style={{width: 200, height: 200}}
                       source={require('../../../assets/logo.png')}/>
                <View>
                    <Label style={text.logoText}>{I18n.t('noConnection')}</Label>
                </View>
            </Container>
        );
    }
}

export default Offline;
