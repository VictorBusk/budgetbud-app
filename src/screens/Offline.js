import React, {Component} from 'react';
import {Container} from "native-base";
import {Logo} from '../components';
import {container} from "../style";
import I18n from "../strings/i18n";

class Offline extends Component {
    render() {
        return (
            <Container style={container.signedOutContainer}>
                <Logo logoText={I18n.t('noConnection')}/>
            </Container>
        );
    }
}

export default Offline;
