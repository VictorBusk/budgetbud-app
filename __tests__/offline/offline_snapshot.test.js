import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Offline from "../../src/app/offline/Offline";

Enzyme.configure({adapter: new Adapter()});

describe('Offline', () => {
    it('renders Offline as expected', () => {
        const wrapper = shallow(
            <Offline/>
        );
        expect(wrapper.dive()).toMatchSnapshot();
    });
});
