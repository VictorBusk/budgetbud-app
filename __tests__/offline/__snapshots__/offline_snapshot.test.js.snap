// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expenses renders ExpenseOverview as expected 1`] = `
<Container
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#03426A",
        "flex": 2,
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
  />
  <View>
    <Styled(Label)
      style={
        Object {
          "color": "rgba(255, 255, 255, 1)",
          "marginBottom": 74.11111111111111,
          "marginTop": 20,
        }
      }
    >
      Lost connection to the internet
    </Styled(Label)>
  </View>
</Container>
`;

exports[`Offline renders Offline as expected 1`] = `
<Container
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "alignItems": "center",
        "backgroundColor": "#03426A",
        "flex": 2,
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Object {
        "height": 200,
        "width": 200,
      }
    }
  />
  <View>
    <Styled(Label)
      style={
        Object {
          "color": "rgba(255, 255, 255, 1)",
          "marginBottom": 74.11111111111111,
          "marginTop": 20,
        }
      }
    >
      Lost connection to the internet
    </Styled(Label)>
  </View>
</Container>
`;
